### DISCUSSION TITLE
Question on action and loader

### DISCUSSION THREAD

**Discussion by rnwonder** [remix-run/remix#9734] [Q&A]
I am curious if there is a way to know on the loader level if it is being called when an action has completed?

For a bit of context I have a loader that handles search base on a search param in the url.

The data is being cached for 5mins for each particular search param.

What I want to do is invalidate the cache after an action on the page succeed so it doesn't get the cache data anymore when loader is called after the action completes. And also it will be nice if it also specify the action url that was successfully called.

**Comment by kiliman** [comment]
Unfortunately, no, but it would be nice if Remix would add a header `x-remix-revalidate: action-route-id`

**Comment by sergiodxa** [comment]
> What I want to do is invalidate the cache after an action on the page succeed so it doesn't get the cache data anymore when loader is called after the action completes

Invalidate the cache in the action, then when the loader runs again it will find the cache is invalid and query/fetch it again.
